### PR TITLE
[libsamplerate] Upgrade to 0.2.0

### PR DESCRIPTION
**Pull request by evpobr**
**Upgrade libsamplerate port to 0.2.0**

* Upgrade to 0.2.0
* Use mainstream CMake project
* Remove obsolete patch
* Add versions to `port_versions/`
* Updated project URL

### CODE REVIEW

**Comment by Be-ing** [comment]
I was about to open a PR for this and saw yours. :laughing:

File: ports/libsamplerate/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "libsamplerate",
-  "version-string": "0.1.9.0",
-  "port-version": 3,
```
Comment (Be-ing) [inline ports/libsamplerate/vcpkg.json:4]: Why did you remove the port-version? Shouldn't it be increased to 4?

File: port_versions/baseline.json [hunk 2]
```diff
@@ -3289,8 +3289,8 @@
       "port-version": 0
     },
     "libsamplerate": {
-      "baseline": "0.1.9.0",
-      "port-version": 3
+      "baseline": "0.2.0",
+      "port-version": 0
```
Comment (Be-ing) [inline port_versions/baseline.json:3293]:
```suggestion
      "port-version": 4
```

**Comment by Be-ing** [comment]
This fixes the macOS build which was broken with 0.1.9.

File: ports/libsamplerate/vcpkg.json (hunk 1 above)
Comment (evpobr) [reply ports/libsamplerate/vcpkg.json:4]: Hmm, may be deleting port-version was correct.

From https://vcpkg.readthedocs.io/en/latest/maintainers/control-files/

> When the version of the underlying library is upgraded, this field should be set back to 0 (i.e., delete the Port-Version field).

Comment (Be-ing) [reply ports/libsamplerate/vcpkg.json:4]: Oh, leave it as is then.

**Comment by Be-ing** [comment]
ping @NancyLi1013

**Comment by Be-ing** [comment]
CI builds failed:
```
Error: Couldn't find explicitly specified baseline `"16002d9c2318dec4c69e02d9af8c0e11dca0d4c6"` in the baseline file, and there was no baseline at that commit or the commit didn't exist.
Error: while checking out baseline '16002d9c2318dec4c69e02d9af8c0e11dca0d4c6:versions/baseline.json':
fatal: Path 'versions/baseline.json' exists on disk, but not in '16002d9c2318dec4c69e02d9af8c0e11dca0d4c6'.
```

**Comment by evpobr** [comment]
One directory was renamed in upstream and pull request is now broken.

**Comment by NancyLi1013** [comment]
@Be-ing and @evpobr 

The current CI failures are caused by our changes.

Don't worry about this. You just need to merge master to this PR.

**Comment by evpobr** [comment]
> @Be-ing and @evpobr
> 
> The current CI failures are caused by our changes.
> 
> Don't worry about this. You just need to merge master to this PR.

@NancyLi1013 , done.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/libsamplerate/portfile.cmake [hunk 3]
```diff
@@ -1,23 +1,25 @@
...
 )
 
-file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH}/src)
-file(COPY ${SOURCE_PATH}/Win32/config.h DESTINATION ${SOURCE_PATH}/src)
-
 vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}/src
+    SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
+    -DBUILD_TESTING=OFF
```
Comment (JonLiu1993) [inline ports/libsamplerate/portfile.cmake:12]: OPTIONS is missing here, and the format is incorrect, you should add OPTIONS and leave four spaces behind

File: ports/libsamplerate/portfile.cmake [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
 
 vcpkg_install_cmake()
 
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/SampleRate TARGET_PATH share/SampleRate)
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES m)
+
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
 
 file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/libsamplerate RENAME copyright)
```
Comment (JonLiu1993) [inline ports/libsamplerate/portfile.cmake:24]: share/libsamplerate should be share/${PORT}

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/libsamplerate/portfile.cmake [hunk 5]
```diff
@@ -1,24 +1,27 @@
...
     PREFER_NINJA
+    OPTIONS
+        -DBUILD_TESTING=OFF
+        -DLIBSAMPLERATE_EXAMPLES=OFF
 )
 
 vcpkg_install_cmake()
 
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/SampleRate TARGET_PATH share/SampleRate)
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES m)
```
Comment (NancyLi1013) [inline ports/libsamplerate/portfile.cmake:20]:
```suggestion
vcpkg_fixup_pkgconfig()
```
This argument has invalided now. So please remove it.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@evpobr ,Please merge the master and run the command vcpkg x-add-version and resubmit

**Comment by evpobr** [comment]
> @evpobr ,Please merge the master and run the command vcpkg x-add-version and resubmit

    PS D:\source\repos\vcpkg> vcpkg x-add-version  
    Error: Use option `--all` to update version files for all ports at once.
    PS D:\source\repos\vcpkg> vcpkg x-add-version --all

Nothing changed.

**Comment by Be-ing** [comment]
:tada:
